resampler/src/speex.rs

```diff
@@ -895,6 +965,59 @@ fn cubic_coef(frac: f32, interp: &mut [f32]) {
             as f32;
 }
 
+#[cfg(feature = "simd")]
+fn resampler_basic_interpolate_double(
+    st: &mut SpeexResamplerState,
+    channel_index: u32,
+    in_0: &[f32],
+    in_len: &mut u32,
+    out: &mut [f32],
+    out_len: &mut u32,
+) -> i32 {
+    let n = st.filt_len as usize;
+    let channel_idx = channel_index as usize;
+    let mut last_sample = st.last_sample[channel_idx];
+    let mut samp_frac_num = st.samp_frac_num[channel_idx];
+    let out_stride = st.out_stride;
+    let int_advance = st.int_advance;
+    let oversample = st.oversample;
+    let frac_advance = st.frac_advance;
+    let den_rate = st.den_rate;
+    let sinc_table = &st.sinc_table;
+
+    let mut out_sample: u32 = 0;
+
+    while !(last_sample >= *in_len || out_sample >= *out_len) {
+        let iptr: &[f32] = &in_0[last_sample as usize..];
+        let offset = samp_frac_num * st.oversample / st.den_rate;
+        let frac =
+            (samp_frac_num * oversample % den_rate) as f32 / den_rate as f32;
+        let mut accum = f64x4::splat(0.0);
+        iptr.iter().zip(0..n).for_each(|(&curr_in, j)| {
+            let idx = (2 + (j + 1) * oversample as usize) - offset as usize;
+            accum += f64x4::from(
```

Comment:
i wonder if its possible to eradicate this duplication since only few lines are changed inside these functions